pacai/core/featureExtractors.py

```diff
@@ -33,8 +81,6 @@ def _closestFood(pos, food, walls):
 
     x1, y1 = pos
 
-    assert not walls[x1][y1], 'pos is a wall: ' + pos
```

Comment:
You can keep this assertion in, just remove the string:
```
assert not walls[x1][y1]
```

The reason for this is purely performance.
Every time this method executes (which would be fairly often), this assertion line would get evaluated.
The assertion expression itself it really quick, just a lookup for a boolean, but the second argument would also get evaluated (even if the assertion does not trigger).
The second argument is an expression, so it would need to be evaluated before it can be passed as an argument to the builtin `assert`.
Unfortunately, this expression contains a string concatenation with an unknown value (pos).
So every call, Python will have to allocate memory for the new string, copy the existing string contents over, and put in pos.
This is super expensive, especially for a method that gets called a lot.

Also, looking at it now, this will not even run.
Instead of an assertion popping, another exception will be thrown:
```
eriq@box:~$ python3
Python 3.7.4 (default, Jul 16 2019, 07:12:58) 
[GCC 9.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> pos = (1, 2)
>>> "pos: " + pos
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate str (not "tuple") to str
>>> 
```